### PR TITLE
docs: add license faq

### PR DESCRIPTION
**Pull request by ZhiHanZ**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

Closes #issue

### CODE REVIEW

File: docs/doc/100-faq/30-license-faqs.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+the Elastic License 2.0 provides a good balance between open-source values and commercial interests.
+Comparing other license such as Business Source License, Custom Community License, Elastic License 2.0 is simple, short and clear.
+There only have three limitations applied:
+1. Cannot provide software as a hosted or managed service with substantial access to features/functionality.
+2. Cannot modify or circumvent license key functionality or remove/obscure protected functionality.
+3. Cannot alter/remove/licensing, copyright, or trademark notices of the licensor in the software.
+
+
+### I would like to reuse some components from the Databend project in my own software, which uses the AGPL or another open source license, Is this possible?
+The Databend team is committed to supporting the open-source community and willing to consider extracting specific internal components that are generally useful as a separate project with its own license, for example APL.
```
Comment (PsiACE) [inline docs/doc/100-faq/30-license-faqs.md:54]: It is confusing here. Personally, I recommend that the licence agreement is strictly followed and no additional promises are made.

File: docs/doc/100-faq/30-license-faqs.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+|---------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Apache 2.0 License  | Core features under the Apache License are free to use and fully open source.                                                                                                                                                                                                                       |
+| Elastic 2.0 License | * Elastic (Free) features are free to use. The source code is available to view and modify under Elastic 2.0 License Limitations  <br> * Elastic (Paid) features require an Enterprise License key to access. The source code is available to view and modify under Elastic 2.0 License Limitations |
```
Comment (PsiACE) [inline docs/doc/100-faq/30-license-faqs.md:24]: It is confusing here. It feels like a distinction needs to be made between Elastic features and Enterprise features. In particular, we mentioned earlier that Databend Enterprise features require a paid license; here, it is noted that we may have some Elastic free features.

**Comment by BohuTANG** [comment]
It looks Vercel not working, can't preview.

File: docs/doc/100-faq/30-license-faqs.md (hunk 2 above)
Comment (ZhiHanZ) [reply docs/doc/100-faq/30-license-faqs.md:24]: Elastic License could provide source avaialble free features, we just need to ensure that all enterprise features are proteced by license key

File: docs/doc/100-faq/30-license-faqs.md (hunk 1 above)
Comment (ZhiHanZ) [reply docs/doc/100-faq/30-license-faqs.md:54]: The promises is made for 3rd party integration solutions for example some close sourced ETL/ELT pipelines which has concern about the license, and need to talk in person

**Comment by BohuTANG** [comment]
```


SyntaxError: /vercel/path0/docs/doc/100-faq/30-license-faqs.md: Expected corresponding JSX closing tag for <br>. (32:202)
--
14:01:39.957 | 30 \| <tr parentName="tbody">
14:01:39.957 | 31 \| <td parentName="tr" {...{"align":null}}>{`Elastic 2.0 License`}</td>
14:01:39.957 | > 32 \| <td parentName="tr" {...{"align":null}}><em parentName="td">{` Elastic (Free) features are free to use. The source code is available to view and modify under Elastic 2.0 License Limitations  `}<br>{` `}</em>{` Elastic (Paid) features require an Enterprise License key to access. The source code is available to view and modify under Elastic 2.0 License Limitations`}</td>
14:01:39.957 | \|                                                                                                                                                                                                           ^
14:01:39.958 | 33 \| </tr>
14:01:39.958 | 34 \| </tbody>
14:01:39.958 | 35 \| </table>


```

**Review by BohuTANG — APPROVED** [review]

**Pull request merged**